docs/Cluster.md

```diff
@@ -0,0 +1,177 @@
+```
+
+The goal is to offload your node from connection and routing table management:
+- incoming connections
+- outgoing connections
+- gossip queries + pings
+- incoming gossip aggregation
+- outgoing gossip dispatch (rebroadcast)
+
+### Prerequisite
+
+You already have a node up and running in a standalone setup (with Bitcoin Core properly configured, etc.).
+
+You know what your `node id` is.
+
+In the following, what we previously called `eclair-node` will be called `backend`. It is to be launched, configured and backed-up exactly like in a standalone setup.
+
+### Minimal/Demo setup
+
+Use this if you want to experiment with the cluster mode on a single local server.
+
+Set the following values in `.eclair/eclair.conf`:
+```
+akka.actor.provider = cluster
+akka.extensions = ["akka.cluster.pubsub.DistributedPubSub"]
+
+// replace this with your node id
+// if you don't know what your node id is, you should probably stop right here
+eclair.front.pub = 03...............
+```
+
+Start the `backend`:
+```shell
+$ ./eclair-node.sh
+```
+
+Then run an instance of `frontend`:
+```shell
+$ ./eclair-front.sh -Dakka.remote.artery.canonical.port=25521 -Declair.server.port=9736
+```
+
+NB: we override the ports, otherwise they would conflict since in this example everything runs on the same server. You can run multiple `frontend`s on the same server, just make sure to change the ports.
+
+### Production setup
+
+In production you should:
+- run multiple `frontend`s
+- run one app per server
+- enable `tcp-tls` to encrypted between members of the cluster with your own generated certificate (see below)
+- use a load balancer to hide all your `frontend` servers under the same ip address
+- set firewall rules to disable lightning connections (port 9735) on your `backend` server, so all connections go through the `frontend`
+- enable [monitoring](Monitoring.md)
+- on AWS, use AWS Secrets Manager (see [AWS deployment](#aws-deployment))
+
+#### Enable encrypted communication for the cluster
+
+We use a self-signed certificate, which offers a good compromise. More advanced options are available, see [akka doc](https://doc.akka.io/docs/akka/current/remoting-artery.html#remote-security).
+> Have a single set of keys and a single certificate for all nodes and disable hostname checking
+> - The single set of keys and the single certificate is distributed to all nodes. The certificate can be self-signed as it is distributed both as a certificate for authentication but also as the trusted certificate.
+> - If the keys/certificate are lost, someone else can connect to your cluster.
+> - Adding nodes to the cluster is simple as the key material can be deployed / distributed to the new node.
+
+Generate a self-signed certificate (set a strong password):
+```shell
+$ keytool -genkeypair -v \
+          -keystore akka-cluster-tls.jks \
+          -dname "O=ACME, C=FR" \
+          -keypass:env <password> \
+          -storepass:env <password> \
+          -keyalg RSA \
```

Comment:
Why do you use RSA and not elliptic curves (which are safer, faster and create smaller certificates)?

Reply:
I used one of the recommended algorithms in [akka doc](https://doc.akka.io/docs/akka/current/remoting-artery.html#configuring-ssl-tls-for-akka-remoting), but advanced users can change if they prefer.